src/app/indexes/Index.js

```diff
@@ -32,10 +37,47 @@ const ensureDocumentsAreValidStructure = (documents, structure) => {
       } catch (e) {
         throw new Error(`${e.message} (document: ${JSON.stringify(document)})`);
       }
-    })
+    });
   });
 };
 
+const attemptToStoreDocuments = async (name, keyFieldName, documents, attempts = 3) => {
+  if (attempts > 0 && documents.length > 0) {
```

Comment:
Could we remove this _if_ block by returning there are no attempts left of documents is empty?

e.g. if (attempts <= 0 || documents.length === 0) return;

Reply:
@AMcCartneyMTH I completely agree to both this and the below point, I've made that update 👍